Sources/BowFree/Program.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+import Bow
+
+/// Witness for the `Program<F, A>` data type. To be used in simulated Higher Kinded Types.
+public final class ForProgram {}
+
+/// Partial application of the `Program` type constructor, omitting the last parameter.
+public final class ProgramPartial<F>: Kind<ForProgram, F> {}
+
+/// Higher Kinded Type alias to improve readability.
+public typealias ProgramOf<F, A> = Kind<ProgramPartial<F>, A>
+
+/// Program is a type that, given any type constructor, is able to provide a Monad instance, that can be interpreted into a more restrictive one.
+///
+/// As opposed to `Free`, `Program` does not require `F` to be a functor, which means that evaluation of `map` calls are deferred
+/// and left for the later interpretation in another monad.
+public final class Program<F, A>: ProgramOf<F, A> {
+    let asFree: Free<CoyonedaPartial<F>, A>
```

Comment:
Would it make sense to expose this as a public field? I'm thinking about a case where we want to implement an instance of a type class for Program and we need to access this.

Reply:
You are right!